src/api/check.rs

```diff
@@ -475,79 +491,51 @@ pub struct Request {
     #[serde(serialize_with = "serialize_option_vec_string")]
     pub enabled_rules: Option<Vec<String>>,
     /// IDs of rules to be disabled, comma-separated.
-    #[cfg_attr(feature = "cli", clap(long))]
     #[serde(serialize_with = "serialize_option_vec_string")]
     pub disabled_rules: Option<Vec<String>>,
     /// IDs of categories to be enabled, comma-separated.
-    #[cfg_attr(feature = "cli", clap(long))]
     #[serde(serialize_with = "serialize_option_vec_string")]
     pub enabled_categories: Option<Vec<String>>,
     /// IDs of categories to be disabled, comma-separated.
-    #[cfg_attr(feature = "cli", clap(long))]
     #[serde(serialize_with = "serialize_option_vec_string")]
     pub disabled_categories: Option<Vec<String>>,
     /// If true, only the rules and categories whose IDs are specified with
     /// `enabledRules` or `enabledCategories` are enabled.
-    #[cfg_attr(feature = "cli", clap(long))]
-    #[serde(skip_serializing_if = "is_false")]
+    #[serde(skip_serializing_if = "std::ops::Not::not")]
     pub enabled_only: bool,
     /// If set to `picky`, additional rules will be activated, i.e. rules that
     /// you might only find useful when checking formal text.
-    #[cfg_attr(
-        feature = "cli",
-        clap(long, default_value = "default", ignore_case = true, value_enum)
-    )]
     #[serde(skip_serializing_if = "Level::is_default")]
     pub level: Level,
 }
 
-impl Default for Request {
-    #[inline]
-    fn default() -> Request {
-        Request {
-            text: Default::default(),
-            data: Default::default(),
+impl<'source> Request<'source> {
+    /// Create a new empty request with language set to `"auto"`.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
             language: "auto".to_string(),
-            username: Default::default(),
-            api_key: Default::default(),
-            dicts: Default::default(),
-            mother_tongue: Default::default(),
-            preferred_variants: Default::default(),
-            enabled_rules: Default::default(),
-            disabled_rules: Default::default(),
-            enabled_categories: Default::default(),
-            disabled_categories: Default::default(),
-            enabled_only: Default::default(),
-            level: Default::default(),
+            ..Default::default()
         }
     }
-}
-
-#[inline]
-fn is_false(b: &bool) -> bool {
-    !(*b)
-}
 
-impl Request {
     /// Set the text to be checked and remove potential data field.
     #[must_use]
-    pub fn with_text(mut self, text: String) -> Self {
-        self.text = Some(text);
+    pub fn with_text<T: Into<Cow<'static, str>>>(mut self, text: T) -> Self {
+        self.text = Some(text.into());
         self.data = None;
         self
     }
 
     /// Set the data to be checked and remove potential text field.
     #[must_use]
-    pub fn with_data(mut self, data: Data) -> Self {
+    pub fn with_data(mut self, data: Data<'static>) -> Self {
```

Comment:
Why do we have static lifetime here? I thought this would be limited to the CLI API.

Reply:
Not sure sorry, I'll remove it